### PR TITLE
DSD-325: add Style Guide category

### PR DESCRIPTION
**Pull request by bigfishdesign13**
Fixes [DSD-325](https://jira.nypl.org/browse/DSD-325)

## **This PR does the following:**
- adds `Style Guide` category to Storybook sidebar

### Front End Review:
- [ ] View [the example in Storybook]()

### CODE REVIEW

File: src/utils/componentCategories.ts [hunk 1]
```diff
@@ -20,6 +20,10 @@ const categories = {
       "Promo",
     ],
   },
+  deprecated: {
+    title: "Deprecated",
+    components: ["CheckboxOld"],
```
Comment (EdwinGuzman) [inline src/utils/componentCategories.ts]: This isn't being displayed. Is it suppose to or is this meant to just be a declaration?

Comment (bigfishdesign13) [reply src/utils/componentCategories.ts]: Michael inserted that when he was reworking `Checkbox`.  We opted to not deprecate the original component, so `CheckboxOld` was never actually created.  However, I think we will eventually want to have a `Deprecated` category, so I left it in there.

**Review by EdwinGuzman — APPROVED** [review]
I like this section in Storybook.

File: src/utils/componentCategories.ts (hunk 1 above)
Comment (bigfishdesign13) [reply src/utils/componentCategories.ts]: You already approved this PR, but I removed `CheckboxOld` from the deprecated section.

**Pull request merged**